File: MultiLineSearch/Settings/PackageMemorySettingsWriter.vb
Imports System
Imports Microsoft.VisualStudio.Shell.Interop
Imports Microsoft.VisualStudio


Namespace Settings

    ''' <summary>
    ''' Writes the package settings into memory.
    ''' </summary>
    ''' <remarks>The settings are stored in <see cref="PackageMemorySettingsStore"/>.</remarks>
    Friend Class PackageMemorySettingsWriter
    Implements IVsSettingsWriter


        Private ReadOnly mSettingsStore As PackageMemorySettingsStore = New PackageMemorySettingsStore

        Public ReadOnly Property SettingsStore() As PackageMemorySettingsStore
            Get
                Return mSettingsStore
            End Get
        End Property


        Public Function ReportError(ByVal pszError As String, ByVal dwErrorType As UInteger) As Integer Implements Microsoft.VisualStudio.Shell.Interop.IVsSettingsWriter.ReportError

        End Function

        Public Function WriteCategoryVersion(ByVal nMajor As Integer, ByVal nMinor As Integer, ByVal nBuild As Integer, ByVal nRevision As Integer) As Integer Implements Microsoft.VisualStudio.Shell.Interop.IVsSettingsWriter.WriteCategoryVersion

        End Function


        Public Function WriteSettingAttribute(ByVal pszSettingName As String, ByVal pszAttributeName As String, ByVal pszSettingValue As String) As Integer Implements Microsoft.VisualStudio.Shell.Interop.IVsSettingsWriter.WriteSettingAttribute
            SettingsStore.SetSettingAttribute(pszSettingName, pszAttributeName, pszSettingValue)
            Return VSConstants.S_OK
        End Function


        Public Function WriteSettingBoolean(ByVal pszSettingName As String, ByVal fSettingValue As Integer) As Integer Implements Microsoft.VisualStudio.Shell.Interop.IVsSettingsWriter.WriteSettingBoolean
            SettingsStore.SetSetting(pszSettingName, fSettingValue)
            Return VSConstants.S_OK
        End Function


        Public Function WriteSettingBytes(ByVal pszSettingName As String, ByVal pSettingValue() As Byte, ByVal lDataLength As Integer) As Integer Implements Microsoft.VisualStudio.Shell.Interop.IVsSettingsWriter.WriteSettingBytes
            Dim newValue As Byte()
            If pSettingValue.Length > lDataLength Then
                newValue = New Byte(lDataLength) {}
                Array.Copy(pSettingValue, 0, newValue, 0, lDataLength)
            Else
                newValue = pSettingValue
            End If
            SettingsStore.SetSetting(pszSettingName, newValue)

            Return VSConstants.S_OK
        End Function


        Public Function WriteSettingLong(ByVal pszSettingName As String, ByVal lSettingValue As Integer) As Integer Implements Microsoft.VisualStudio.Shell.Interop.IVsSettingsWriter.WriteSettingLong
            SettingsStore.SetSetting(pszSettingName, lSettingValue)
            Return VSConstants.S_OK
        End Function


        Public Function WriteSettingString(ByVal pszSettingName As String, ByVal pszSettingValue As String) As Integer Implements Microsoft.VisualStudio.Shell.Interop.IVsSettingsWriter.WriteSettingString
            SettingsStore.SetSetting(pszSettingName, pszSettingValue)
            Return VSConstants.S_OK
        End Function


        Public Function WriteSettingXml(ByVal pIXMLDOMNode As Object) As Integer Implements Microsoft.VisualStudio.Shell.Interop.IVsSettingsWriter.WriteSettingXml

        End Function

        Public Function WriteSettingXmlFromString(ByVal szXML As String) As Integer Implements Microsoft.VisualStudio.Shell.Interop.IVsSettingsWriter.WriteSettingXmlFromString

        End Function


    End Class

End Namespace

File: MultiLineSearch2/Settings/PackageMemorySettingsWriter.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using Microsoft.VisualStudio.Shell.Interop;
using Microsoft.VisualStudio;


namespace Helixoft.MultiLineSearch.Settings
{

    /// <summary>
    /// Writes the package settings into memory.
    /// </summary>
    /// <remarks>The settings are stored in <see cref="PackageMemorySettingsStore"/>.</remarks>
    internal class PackageMemorySettingsWriter : IVsSettingsWriter
    {



        private readonly PackageMemorySettingsStore mSettingsStore = new PackageMemorySettingsStore();
        public PackageMemorySettingsStore SettingsStore
        {
            get { return mSettingsStore; }
        }


        public int ReportError(string pszError, uint dwErrorType)
        {
            return VSConstants.S_OK;
        }

        public int WriteCategoryVersion(int nMajor, int nMinor, int nBuild, int nRevision)
        {
            return VSConstants.S_OK;
        }


        public int WriteSettingAttribute(string pszSettingName, string pszAttributeName, string pszSettingValue)
        {
            SettingsStore.SetSettingAttribute(pszSettingName, pszAttributeName, pszSettingValue);
            return VSConstants.S_OK;
        }


        public int WriteSettingBoolean(string pszSettingName, int fSettingValue)
        {
            SettingsStore.SetSetting(pszSettingName, fSettingValue);
            return VSConstants.S_OK;
        }


        public int WriteSettingBytes(string pszSettingName, byte[] pSettingValue, int lDataLength)
        {
            byte[] newValue = null;
            if (pSettingValue.Length > lDataLength)
            {
                newValue = new byte[lDataLength + 1];
                Array.Copy(pSettingValue, 0, newValue, 0, lDataLength);
            }
            else
            {
                newValue = pSettingValue;
            }
            SettingsStore.SetSetting(pszSettingName, newValue);

            return VSConstants.S_OK;
        }


        public int WriteSettingLong(string pszSettingName, int lSettingValue)
        {
            SettingsStore.SetSetting(pszSettingName, lSettingValue);
            return VSConstants.S_OK;
        }


        public int WriteSettingString(string pszSettingName, string pszSettingValue)
        {
            SettingsStore.SetSetting(pszSettingName, pszSettingValue);
            return VSConstants.S_OK;
        }


        public int WriteSettingXml(object pIXMLDOMNode)
        {
            return VSConstants.S_OK;
        }

        public int WriteSettingXmlFromString(string szXML)
        {
            return VSConstants.S_OK;
        }


    }

}
